webui/react/src/hooks/useCustomizeColumnsModal.tsx

```diff
@@ -0,0 +1,234 @@
+import { Button, Input, Modal, ModalFuncProps } from 'antd';
+import { ModalFunc } from 'antd/es/modal/confirm';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { FixedSizeList as List } from 'react-window';
+
+import { isEqual } from 'utils/data';
+import { camelCaseToSentence } from 'utils/string';
+
+import css from './useCustomizeColumnsModal.module.scss';
+
+interface ModalState {
+  columns: string[];
+  defaultVisibleColumns: string[];
+  initialVisibleColumns: string[];
+  onSave?: (columns: string[]) => void;
+  visible: boolean;
+}
+
+export interface ShowModalProps {
+  columns: string[];
+  defaultVisibleColumns: string[];
+  initialVisibleColumns: string[];
+  onSave?: (columns: string[]) => void;
+}
+
+interface ModalHooks {
+  showModal: (props: ShowModalProps) => void;
+}
```

Comment:
**suggestion:** inherit the `useModal ModalHooks` definition. Something like:
```
interface ModalHooks extends Omit<Hooks, 'modalOpen'> {
  modalOpen: (props: ShowModalProps) => void;
}
```

goal of providing a consistent modal hook interface, even though the props between the functions might vary. So ensuring that the `modalClose` and `modalRef` are returned in case they are needed.